spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkMLTrainer.java

```diff
@@ -0,0 +1,152 @@
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Common class for Spark Trainers. Contains common Spark ML Trainer properties, configuartion ,validation and
+ * feature extraction methods to be used in trainers.
+ */
+public abstract class SparkMLTrainer extends SparkSink<StructuredRecord> {
+
+  private MLTrainerConfig config;
+  private static final Logger LOG = LoggerFactory.getLogger(SparkMLTrainer.class);
+
+  /**
+   * Config class for Trainers. Contains common config properties to be used in trainers.
+   */
+  protected class MLTrainerConfig extends PluginConfig {
+
+    @Description("The name of the FileSet to save the model to.")
+    protected String fileSetName;
+
+    @Description("Path of the FileSet to save the model to.")
+    protected String path;
+
+    @Nullable
+    @Description("A comma-separated sequence of fields that needs to be used for training.")
+    protected String featureFieldsToInclude;
+
+    @Nullable
+    @Description("A comma-separated sequence of fields that needs to be excluded from being used in training.")
+    protected String featureFieldsToExclude;
+
+    @Description("The field from which to get the label. It must be of type double.")
+    protected String labelField;
+
+    protected MLTrainerConfig() {
+    }
+
+    protected MLTrainerConfig(String fileSetName, String path, @Nullable String featureFieldsToInclude,
+                           @Nullable String featureFieldsToExclude, String labelField) {
+      this.fileSetName = fileSetName;
+      this.path = path;
+      this.featureFieldsToInclude = featureFieldsToInclude;
+      this.featureFieldsToExclude = featureFieldsToExclude;
+      this.labelField = labelField;
+    }
+
+    protected void validate(Schema inputSchema) {
+      SparkUtils.validateConfigParameters(inputSchema, featureFieldsToInclude, featureFieldsToExclude, labelField,
+                                          null);
+      SparkUtils.validateLabelFieldForTrainer(inputSchema, labelField);
+    }
+  }
+
+  public SparkMLTrainer(MLTrainerConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    pipelineConfigurer.createDataset(config.fileSetName, FileSet.class);
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    Preconditions.checkArgument(inputSchema != null, "Input Schema must be a known constant.");
+    config.validate(inputSchema);
+  }
+
+  @Override
+  public void prepareRun(SparkPluginContext context) throws Exception { }
+
+  protected Map<String, Integer> getFeatures(Schema inputSchema, @Nullable String featuresToInclude,
+                                             @Nullable String featuresToExclude, String labelField) {
+    return SparkUtils.getFeatureList(inputSchema, featuresToInclude, featuresToExclude, labelField);
+  }
+
+  public static JavaRDD<LabeledPoint> getTrainingData(final Map<String, Integer> fields,
```

Comment:
Please refactor DecisionTreeTrainer to extend this class to validate that it works as expected and make sure there isn't anything missing.